js/content/community.js

```diff
@@ -423,17 +423,44 @@ let ProfileHomePageClass = (function(){
                 </div>`;
         }
 
-        // profile permalink
-        if (SyncedStorage.get("profile_permalink")) {
-            let imgUrl = ExtensionLayer.getLocalUrl("img/clippy.svg");
-            htmlstr +=
-                `<div id="es_permalink_div" class="profile_count_link">
-					<span class="count_link_label">${Localization.str.permalink}</span>
-					<div class="es_copy_wrap">
-						<input id="es_permalink" type="text" value="https://steamcommunity.com/profiles/${steamId}" readonly />
-						<button id="es_permalink_copy"><img src="${imgUrl}" /></button>
-					</div>
-				</div>`;
+        // profile steamid
+        if (SyncedStorage.get("profile_steamid")) {
+            let dropdown = document.querySelector("#profile_action_dropdown .popup_body.popup_menu");
+            if (dropdown) {
+                dropdown.innerHTML += 
+                    `<a class="popup_menu_item" id="es_steamid">
+                        <img src="https://steamcommunity-a.akamaihd.net/public/images/skin_1/iconForums.png">&nbsp; ${Localization.str.view_steamid}
+                    </a>`;
+            } else {
+                let actions = document.querySelector(".profile_header_actions");
+                if (actions) {
+                    actions.innerHTML +=
+                        `<a class="btn_profile_action btn_medium" id="es_steamid">
+                            <span>${Localization.str.view_steamid}</span>
+                        </a>`;
+                }
+            }
+
+            document.querySelector("#es_steamid").addEventListener("click", showSteamIdDialog);
+            ExtensionLayer.runInPageContext(`function() {
+                $J(document).on("click", ".es_copy", function(event) {
+                    let elem = $J(event.target).closest(".es_copy");
+                    let text = elem.data("copy");
+                    setClipboard(text);
+                    window.idDialog.Dismiss();
+                    ShowAlertDialog(g_rgProfileData.personaname + "'s SteamID", \`${Localization.str.copied}\`.replace("__text__", text));
+                });
+
+                function setClipboard(content) {
+                    // Based on https://stackoverflow.com/a/12693636
```

Comment:
Can this be abstracted out to something in common.js or core.js? You've written this exact function in a few places and hopefully in a couple of years it can be replaced with the official HTML5 Clipboard API when that's more supported (https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API#Clipboard).

Reply:
Done